tests/testConstraintManifold.cpp

```diff
@@ -0,0 +1,195 @@
+      std::bind(&ConstraintManifold::recover<Pose3>, std::placeholders::_1,
+                x3_key, boost::none);
+  EXPECT(assert_equal(numericalDerivative11<Pose3, ConstraintManifold, 6>(
+                          x3_recover_func, manifold),
+                      H_recover_x3));
+}
+
+// cart-pole robot setting
+auto robot = CreateRobotFromFile(kUrdfPath + std::string("cart_pole.urdf"))
+                  .fixLink("l0");
+int j0_id = robot.joint("j0")->id(), j1_id = robot.joint("j1")->id();
+int l1_id = robot.link("l1")->id(), l2_id = robot.link("l2")->id(),
+    l0_id = robot.link("l0")->id();
+
+
+std::string pose_str(const Pose3& pose) {
+  const Point3 trans = pose.translation();
+  const Rot3 rot = pose.rotation();
+  const Vector3 xyz = rot.xyz();
+  std::stringstream buffer;
+  buffer << boost::format(
+                "[t: [%0.4f, %0.4f, %0.4f],\tR: [%0.4f, %0.4f, %0.4f]]") %
+                trans.x() % trans.y() % trans.z() % xyz(0) % xyz(1) % xyz(2);
+  return buffer.str();
+}
+
+std::string vec_str(const Vector& vec) {
+  std::stringstream buffer;
+  buffer << boost::format("[%0.4f, %0.4f, %0.4f, %0.4f, %0.4f, %0.4f]") %
+                vec(0) % vec(1) % vec(2) % vec(3) % vec(4) % vec(5);
+  return buffer.str();
+}
+
+
+void print_state(const Values& values) {
+  const Pose3 pose1 = Pose(values, l1_id), pose2 = Pose(values, l2_id);
+  const Vector6 twist1 = Twist(values, l1_id), twist2 = Twist(values, l2_id);
+  const Vector6 twista1 = TwistAccel(values, l1_id),
+                twista2 = TwistAccel(values, l2_id);
+  const Vector6 wrench11 = Wrench(values, l1_id, j0_id),
+                wrench01 = Wrench(values, l0_id, j0_id),
+                wrench12 = Wrench(values, l1_id, j1_id),
+                wrench22 = Wrench(values, l2_id, j1_id);
+  const double q1 = JointAngle(values, j0_id), q2 = JointAngle(values, j1_id);
+  const double v1 = JointVel(values, j0_id), v2 = JointVel(values, j1_id);
+  const double a1 = JointAccel(values, j0_id), a2 = JointAccel(values, j1_id);
+  const double t1 = Torque(values, j0_id), t2 = Torque(values, j1_id);
+  std::cout << "q: " << boost::format("(%0.4f)") % q1 << "\t" << pose_str(pose1)
+            << "\t" << boost::format("(%0.4f)") % q2 << "\t" << pose_str(pose2)
+            << "\n";
+  std::cout << "v: " << boost::format("(%0.4f)") % v1 << "\t" << vec_str(twist1)
+            << "\t" << boost::format("(%0.4f)") % v2 << "\t" << vec_str(twist2)
+            << "\n";
+  std::cout << "a: " << boost::format("(%0.4f)") % a1 << "\t"
+            << vec_str(twista1) << "\t" << boost::format("(%0.4f)") % a2 << "\t"
+            << vec_str(twista2) << "\n";
+  std::cout << "f1: " << vec_str(wrench01) << boost::format("(%0.4f)") % t1
+            << "\t" << vec_str(wrench11) << "\n";
+  std::cout << "f2: " << vec_str(wrench12) << boost::format("(%0.4f)") % t2
+            << "\t" << vec_str(wrench22) << "\n";
+}
+
+TEST(ConstraintManifold, Kinematics) {
```

Comment:
What is this test doing?

Reply:
Temporarily removing this test, replaced with a TODO, will add back later.

Reply:
I would rather you don't add the TODO and just get it done now. That way it'll be off your plate forever.

Reply:
Done